docs/advanced/offline.md

```diff
@@ -0,0 +1,157 @@
+
+There is also a [`useOnlineStatus`](#online-status) hook which returns the online or offline status of the client.
+
+To see a good example of these functions' APIs and their usage, see `SectionWrapper.js` in the [PWA example app](https://github.com/dhis2/app-platform/tree/master/examples/pwa-app/src/components/SectionWrapper.js) in the platform repository.
+
+## Cacheable sections
+
+> This feature can only be used when PWA is enabled in `d2.config.js`. See the [App Platform docs](https://platform.dhis2.nu/#/pwa/pwa) for more information.
+
+These features are supported by an `<OfflineProvider>` component which the app platform provides to the app.
+
+### How it works
+
+Cacheable sections enable sections of an app to be individually cached offline on demand. Using the `CacheableSection` wrapper and the `useCacheableSection` hook, when a user requests a section to be cached for offline use, the section's component tree will rerender, and the app's service worker will listen to all the network traffic for the component to cache it offline. To avoid caching that components' data before a user requests to do so, you can use the [URL filters feature](https://platform.dhis2.nu/#/pwa/pwa?id=opting-in) in `d2.config.js`.
+
+Note that, without using these features, an app using offline caching will cache all the data that is requested by user as they use the app without needing to use cacheable sections.
+
+Keep an eye out for this feature in use in the Dashboards app coming soon!
+
+### Usage
+
+Wrap the component to be cached in a `CacheableSection` hook, providing an `id` and `loadingMask` prop. The loading mask should block the screen from interaction, and it will be rendered while the component is in 'recording mode'. Note that the `useCacheableSection` hook _does not need to be used in the same component_ as the `<CacheableSection>`, they just need to use the same `id`. There is an example of this in the file linked below.
+
+Here is an example of the basic usage:
+
+```jsx
+import { CacheableSection, useCacheableSection } from '@dhis2/app-runtime'
+import { Button, Layer, CenteredContent, CircularLoader } from '@dhis2/ui'
+import { Dashboard } from './Dashboard'
+
+/** An example loading mask */
+const LoadingMask = () => (
+    <Layer translucent>
+        <CenteredContent>
+            <CircularLoader />
+        </CenteredContent>
+    </Layer>
+)
+
+export function CacheableSectionWrapper({ id }) {
+    const { startRecording, isCached, lastUpdated } = useCacheableSection(id)
+
+    return (
+        <div>
+            <p>{isCached ? `Last updated: ${lastUpdated}` : 'Not cached'}</p>
+            <Button onClick={startRecording}>
+            <CacheableSection id={id} loadingMask={<LoadingMask />}>
+                <Dashboard id={id} />
+            </CacheableSection>
+        </div>
+    )
+}
+```
+
+#### `CacheableSection` API
```

Comment:
We shouldn't add these docs yet

Reply:
Oops, I meant to delete all but the online status 🤭